compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIESplitLogicalObjFifos.cpp

```diff
@@ -390,12 +390,37 @@ void AMDAIESplitLogicalObjFifosPass::runOnOperation() {
                               "correctly split logical objectFifos.";
     return signalPassFailure();
   }
+  // Use the maximum number of columns available on the device for the default
+  // split factor.
   int64_t numColumns = maybeNumColumns.value();
 
+  // Check if any CoreOp is present, and if so, get the number of columns
+  // actually in use.
+  std::optional<int64_t> mayNumColumnsInUse;
```

Comment:
It would be useful to create a utility function for this and even put it in `AMDAIEUtils` as it might be useful in other places as well in the future.

Also, I think you don't need to make it an optional you could calculate the number of columns as `columnsInUse = maxColumnInUse - minColumnInUse` with both max and min initialized to `0`. The 'special' case would then be when `columnsInUse  == 0`.